db/schema_tables.cc

```diff
@@ -1095,26 +1166,88 @@ future<> store_column_mapping(distributed<service::storage_proxy>& proxy, schema
 // and or filesystem calls, e.g. fsync.
 constexpr size_t max_concurrent = 8;
 
+// Extracts the names of tables affected by a schema mutation.
+// The mutation must target one of the tables in schema_tables_holding_schema_mutations().
+static
+table_selector get_affected_tables(const sstring& keyspace_name, const mutation& m) {
+    const schema& s = *m.schema();
+    auto get_table_name = [&] (const clustering_key& ck) {
+        // The first component of the clustering key in each table listed in
+        // schema_tables_holding_schema_mutations contains the table name.
+        return value_cast<sstring>(utf8_type->deserialize(ck.get_component(s, 0)));
+    };
+    table_selector result;
+    if (m.partition().partition_tombstone()) {
+        slogger.trace("Mutation of {}.{} for keyspace {} contains a partition tombstone",
+                      m.schema()->ks_name(), m.schema()->cf_name(), keyspace_name);
+        result.all_in_keyspace = true;
+    }
+    for (auto&& e : m.partition().row_tombstones()) {
+        const range_tombstone& rt = e.tombstone();
+        if (rt.start.size(s) == 0 || rt.end.size(s) == 0) {
+            slogger.trace("Mutation of {}.{} for keyspace {} contains a multi-table range tombstone",
+                          m.schema()->ks_name(), m.schema()->cf_name(), keyspace_name);
+            result.all_in_keyspace = true;
+            break;
+        }
+        auto table_name = get_table_name(rt.start);
+        if (table_name != get_table_name(rt.end)) {
```

Comment:
Minor, but I think folding the table comparison into the previous `if` would make the code a little clearer. Similar to testing `!some_optional || *some_optional != ...`.

Reply:
It would double increase the calls to get_table_name() through.